core/src/main/java/org/mvcspec/ozark/core/ViewResponseFilter.java

```diff
@@ -111,6 +116,7 @@ public void filter(ContainerRequestContext requestContext,
 
         final Method method = resourceInfo.getResourceMethod();
         final Class<?> returnType = method.getReturnType();
+        final String defaultExtension = ozarkConfig.getDefaultViewFileExtension();
```

Comment:
Perhaps we should let `appendExtensionIfRequired` lookup the default extension instead of doing this here? This way we won't need to pass this into every call to `appendExtensionIfRequired`. I think that's fine because effectively `appendExtensionIfRequired` will only be called one per request so there won't be any performance befit of reading the value here.

The only argument for doing the lookup of the default value outside of `appendExtensionIfRequired` is that it simplifies unit testing. But we could work around this by doing something like this:

```java
// call this method in the filter
private static String appendExtensionIfRequired(String viewName) {
  return appendExtensionIfRequired(viewName, ozarkConfig.getDefaultViewFileExtension());
}

protected static String appendExtensionIfRequired(String viewName, String defaultExtension) {
  // same as in your pull request
}
```

Reply:
Done.